app/models/automation_request.rb

```diff
@@ -29,31 +29,17 @@ def self.create_from_ws(version, user, uri_parts, parameters, requester)
     options[:schedule_type] = parameters['schedule_time'].present? ? "schedule" : "immediately"
     options[:schedule_time] = parameters['schedule_time'].to_i.days.from_now if parameters['schedule_time']
 
-    object_parameters = parse_out_objects(parameters)
-    attrs = MiqRequestWorkflow.parse_ws_string(parameters)
-    attrs.merge!(object_parameters)
-
-    attrs[:userid]     = user.userid
     options[:user_id]  = user.id
-    options[:attrs]    = attrs
+    options[:attrs]    = build_attrs(parameters, user)
     options[:miq_zone] = zone(options) if options[:attrs].key?(:miq_zone)
 
     create_request(options, user, auto_approve)
   end
 
   def self.create_from_scheduled_task(user, uri_parts, parameters)
-    [:namespace, :class_name].each { |key| uri_parts.delete(key) if uri_parts.key?(key) }
-    approval = {'auto_approve' => true}
-    uri_parts.stringify_keys!
-    parameters.stringify_keys!
-    create_from_ws("1.1", user, uri_parts, parameters, approval)
-  end
-
-  def self.parse_out_objects(parameters)
-    object_hash = parameters.select { |key, _v| key.to_s.include?(MiqAeEngine::MiqAeObject::CLASS_SEPARATOR) }
-    object_hash.each do |key, _v|
-      parameters.delete(key)
-    end
+    uri_parts = uri_parts.except(:namespace, :class_name).stringify_keys
```

Comment:
In this case we can use `stringify_keys!` because `except` is already creating a new object.

Reply:
Mmm?
```ruby
NoMethodError:
       undefined method `strip' for :namespace:Symbol
     # ./app/models/miq_request_workflow.rb:364:in `block in parse_request_parameter_hash'
     # ./app/models/miq_request_workflow.rb:362:in `each'
     # ./app/models/miq_request_workflow.rb:362:in `each_with_object'
     # ./app/models/miq_request_workflow.rb:362:in `parse_request_parameter_hash'
```